src/azure-cli-core/azure/cli/core/_help.py

```diff
@@ -122,6 +189,116 @@ def _load_from_data(self, data):
+        if not info:
+            return data
+
+        new_data["type"] = info_type
+
+        if "summary" in info:
+            new_data["short-summary"] = info["summary"]
+
+        if "description" in info:
+            new_data["long-summary"] = info["description"]
+
+        if "examples" in info:
+            new_data["detailed_examples"] = info["examples"]
+            reg_examples = []
+            for item in new_data["detailed_examples"]:
+                ex = {}
+                ex["name"] = item.get("summary", "")
+                ex["text"] = item.get("description", "")
+                ex["text"] = "{}\n{}".format(ex["text"], item.get("command", "")) if ex["text"] else item.get("command", "")
+                ex["min_profile"] = item.get('min_profile')
+                ex["max_profile"] = item.get('max_profile')
+                reg_examples.append(ex)
+            new_data["examples"] = reg_examples
+
+        if "links" in info:
+            text = self._get_links_as_text(info["links"])
+            new_data["links"] = info["links"]
+            new_data["long-summary"] = "{}\n{}".format(new_data["long-summary"], text) \
+                if new_data.get("long-summary") else text
+
+        if "arguments" in info:
+            new_data["parameters"] = []
+            for arg_info in info["arguments"]:
+                new_data["parameters"].append(self._get_parameter_info(arg_info))
+
+        return new_data
+
+    @staticmethod
+    def _get_links_as_text(links):
+        text = ""
+        for link in links:
+            if "name" in link and "url" in link:
+                text += "- {}: {}.\n".format(link["name"], link["url"])
+            elif "url" in link:
+                text += "- {}.\n".format(link["url"])
+        return text
+
+    @staticmethod
+    def _get_parameter_info(arg_info):
+        params = {}
+
+        # only update if new information
+        if "name" in arg_info:
+            params["name"] = arg_info["name"]
+
+        if "summary" in arg_info:
+            params["short-summary"] = arg_info["summary"]
+
+        if "description" in arg_info:
+            params["long-summary"] = arg_info["description"]
+
+        if "value-source" in arg_info:
+            value_source = []
+            for item in arg_info["value-source"]:
+                if "string" in item:
+                    value_source.append(item["string"])
+                elif "link" in item:
+                    link_text = "{}".format(item["link"].get("text", ""))
+                    if "command" in item["link"]:
+                        link_text = "{} command: {} ".format(link_text, item["link"]["command"])
+                    if "url" in item["link"]:
+                        link_text = "{} info: {} ".format(link_text, item["link"]["url"])
+                    value_source.append(link_text.strip())
+            params["populator-commands"] = value_source
+
+        return params
```

Comment:
Right now, it looks like this is just copying the new fields into the old fields and relying on the logic that parses the old fields. How much more difficult would it be to actually have this conditional logic in the places that parse the YAML dictionary into the Help objects?  This seems like it would be limiting in the long run.